### PR TITLE
feat: pieces versioning

### LINKED ISSUE #714: Update Frontend to Read New Pieces

### PR DESCRIPTION
**Pull request by khaledmashaly**
## What does this PR do?

Introduce pieces versioning :star_struck: 

Fixes #714 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/edit-step-accodion.component.ts [hunk 1]
```diff
@@ -126,10 +137,15 @@ export class EditStepAccordionComponent implements AfterViewInit {
...
       name: stepSelected.name,
+      version: (stepSelected.settings as PieceActionSettings).pieceVersion,
     });
     const inputControl = this.stepForm.get('input')!;
     const settings = stepSelected.settings;
-    inputControl.setValue({ ...settings, type: stepSelected.type });
+    inputControl.setValue({
+      ...settings,
+      type: stepSelected.type,
+      teeeeeeeeeeeeeeeeest: 'teeeeeeeeeest',
```
Comment (abuaboud) [inline packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/edit-step-accodion.component.ts]: I think this is by mistake

**Review by abuaboud — APPROVED** [review]

File: packages/backend/src/app/pieces/pieces.controller.ts [hunk 2]
```diff
@@ -34,12 +29,12 @@ export const piecesController: FastifyPluginAsync = async (app) => {
         }
     );
 
-    app.get("/v2/pieces", async () => {
+    app.get("/v1/pieces", async () => {
```
Comment (abuaboud) [inline packages/backend/src/app/pieces/pieces.controller.ts:32]: Why not keep it /v1/ no one is using it?

**Review by abuaboud — APPROVED** [review]

File: packages/backend/src/app/pieces/pieces.controller.ts (hunk 2 above)
Comment (khaledmashaly) [reply packages/backend/src/app/pieces/pieces.controller.ts:32]: It's version one already.

File: packages/frontend/src/app/modules/common/components/form-controls/interpolating-text-form-control/mentions-list/piece-step-mention-item/piece-step-mention-item.component.ts [hunk 3]
```diff
@@ -104,77 +104,75 @@ export class PieceStepMentionItemComponent implements OnInit {
...
+              ? pieceMetadata.triggers[step.settings.triggerName].sampleData ??
+                  {}
+              : {};
+          } else {
+            return step.settings.actionName
+              ? pieceMetadata.actions[step.settings.actionName].sampleData ?? {}
+              : {};
           }
-        }
-        throw new Error("Activepieces- step isn't of a piece type");
```
Comment (AbdulTheActivePiecer) [inline packages/frontend/src/app/modules/common/components/form-controls/interpolating-text-form-control/mentions-list/piece-step-mention-item/piece-step-mention-item.component.ts:127]: why was this removed?

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-action-input-form/piece-action-input-form.component.ts [hunk 4]
```diff
@@ -151,26 +153,22 @@ export class PieceActionInputFormComponent
...
-        const actionsKeys = Object.keys(component.actions);
+
+    this.actions$ = pieceMetadata$.pipe(
+      map((pieceMetadata) => {
+        const actionsKeys = Object.keys(pieceMetadata.actions);
         return actionsKeys.map((actionName) => {
-          const action = component.actions[actionName];
+          const action = pieceMetadata.actions[actionName];
 
           const configs = Object.entries(action.props).map((keyEntry) => {
```
Comment (AbdulTheActivePiecer) [inline packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-action-input-form/piece-action-input-form.component.ts]: can you please make keyEntry => [propName,prop] ? sorry I should have done that at the start.

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-trigger-input-form/piece-trigger-input-form.component.ts [hunk 5]
```diff
@@ -137,7 +135,7 @@ export class PieceTriggerInputFormComponent {
           const configs = Object.entries(trigger.props).map((keyEntry) => {
```
Comment (AbdulTheActivePiecer) [inline packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-trigger-input-form/piece-trigger-input-form.component.ts]: can you please make keyEntry => [propName,prop] ? sorry I should have done that at the start.

File: packages/frontend/src/app/modules/common/components/form-controls/interpolating-text-form-control/mentions-list/piece-step-mention-item/piece-step-mention-item.component.ts (hunk 3 above)
Comment (khaledmashaly) [reply packages/frontend/src/app/modules/common/components/form-controls/interpolating-text-form-control/mentions-list/piece-step-mention-item/piece-step-mention-item.component.ts:127]: fixed in 89e347d

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-action-input-form/piece-action-input-form.component.ts (hunk 4 above)
Comment (khaledmashaly) [reply packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-action-input-form/piece-action-input-form.component.ts]: fixed in 6ce6500

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-trigger-input-form/piece-trigger-input-form.component.ts (hunk 5 above)
Comment (khaledmashaly) [reply packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-trigger-input-form/piece-trigger-input-form.component.ts]: fixed in 6ce6500

**Review by abuaboud — DISMISSED** [review]
